addons/io/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/NeeoService.java

```diff
@@ -0,0 +1,555 @@
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.io.neeo;
+
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import javax.servlet.ServletException;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.binding.BindingInfoRegistry;
+import org.eclipse.smarthome.core.events.Event;
+import org.eclipse.smarthome.core.events.EventFilter;
+import org.eclipse.smarthome.core.events.EventPublisher;
+import org.eclipse.smarthome.core.events.EventSubscriber;
+import org.eclipse.smarthome.core.items.ItemRegistry;
+import org.eclipse.smarthome.core.items.events.ItemStateEvent;
+import org.eclipse.smarthome.core.net.NetworkAddressService;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.link.ItemChannelLinkRegistry;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry;
+import org.eclipse.smarthome.core.thing.type.ThingTypeRegistry;
+import org.eclipse.smarthome.io.transport.mdns.MDNSClient;
+import org.openhab.io.neeo.internal.NeeoDeviceDefinitions;
+import org.openhab.io.neeo.internal.NeeoServlet;
+import org.openhab.io.neeo.internal.NeeoUtil;
+import org.openhab.io.neeo.internal.ServiceContext;
+import org.openhab.io.neeo.internal.discovery.BrainDiscovery;
+import org.openhab.io.neeo.internal.discovery.DiscoveryListener;
+import org.openhab.io.neeo.internal.discovery.FixedBrainDiscovery;
+import org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery;
+import org.openhab.io.neeo.internal.models.NeeoSystemInfo;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The main entry point for the tranport service. The transport service will listen for brain broadcasts and create a
```

Comment:
typo: tranport -> transport

Reply:
@Hilbrand 
Thanks - but question for you.  Is this a connector or a transport?  I've seen verbage both ways for things similar to this.  I've changed these instances to connector to be consistent with everything in it - but I think transport is probably a better term.  Any strong opinions?

Reply:
I have no specific opinion. My knowledge of openHAB doens't go that far :smile: . transport seems to refer to a protocol and connector to the object connecting to some a device translating the device protocol to/from openHAB.